app/models/match.rb

```diff
@@ -17,23 +17,10 @@
 #
 
 class Match < ApplicationRecord
-  has_many :home_matches, as: :team_home, class_name: 'Match'
-  has_many :away_matches, as: :team_away, class_name: 'Match'
-  def matches
-    home_matches.or away_matches
-  end
   belongs_to :team_home, polymorphic: true
   belongs_to :team_away, polymorphic: true
   belongs_to :event, dependent: :delete
 
```

Comment:
Maybe add a `has_many :match_results` or something to have references both ways (and have deletion/updates cascade correctly).